src/Zomp.SyncMethodGenerator/SyncMethodSourceGenerator.cs

```diff
@@ -77,6 +102,71 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
             });
     }
 
+    private static UserMappings GetUserMappings(ImmutableArray<(string Path, string Content)> array, CancellationToken token)

```

Comment:
nit 

may be it would be better to move this this inside an `UserMappings` class and make public and change it name to `Create`?